Take screenshots

Superdavo0001 opened the pull request:
Adds a feature which allows screenshots to be taken at arbitrary resolutions, instead of having to capture the window externally or extract a frame from a video. This branch also fixes a small bug where the reset camera button appears when encoding with resolution-independent zoom (or the zoom multiplier set to something other than 1), and abbreviates the name to res-independent zoom in both the screenshot & video recording windows to keep it from being too long to fit in the window. The now-redundant Recording1080p variable is also removed.

Superdavo0001 committed:
Added Screen Capture

+ Screen capture, can be selected in settings->capture screens
* Fixed a minor bug where the hit test setting in track recording would conserve this setting after recording is completed

Superdavo0001 committed:
Fixed illegal default track name screenshot

It now just saves as "Untitled Track"

Superdavo0001 committed:
Removed comment clutter

Just cleans up the code a bit, it should have no functional effect on the program itself

Superdavo0001 committed:
Fix user camera sprite

Fixed the button to reset the camera appearing whenever the zoom multiplier isn't set to 1, including when recording with resolution-independent zoom. Also abbreviated the name to res-independent zoom.

RatherBeLunar commented:
@Superdavo0001 Not sure if this is intentional or something you've noticed already, but resolution independent zoom doesn't seem to get applied to the UI if you include it in a screenshot larger than your window. This is 8000x4500, my screen is 1920x1080.
![Untitled Track](https://user-images.githubusercontent.com/19538060/94215177-c0745900-fea9-11ea-965e-a159035e0659.png)

Superdavo0001 commented:
Well spotted, I'd completely missed that. I guess it's only a minor issue but I'll get it fixed

Superdavo0001 committed:
Fix reset camera button + UI rescaling

• The UI is now rescaled if resolution-independent zoom is enabled
• The reset camera button now no longer appears during zoom transitions (I think this was a floating-point based error)
• Apparently the code didn't actually check for the res-independent zoom setting, so the code was assuming it to be set. That's now fixed.

RatherBeLunar committed:
Merge branch 'master' into TakeScreenshots
